WordPress/Classes/Services/CommentService.m

```diff
@@ -992,8 +992,10 @@ - (RemoteComment *)remoteCommentWithComment:(Comment *)comment
 {
     id<CommentServiceRemote>remote;
     // TODO: refactor API creation so it's not part of the model
-    if (blog.restApi) {
-        remote = [[CommentServiceRemoteREST alloc] initWithApi:blog.restApi siteID:blog.dotComID];
+    if ([blog supports:BlogFeatureWPComRESTAPI]) {
+        if (blog.restApi) {
+            remote = [[CommentServiceRemoteREST alloc] initWithApi:blog.restApi siteID:blog.dotComID];
```

Comment:
Couldn't the init method return nil if the API parameter was nil? The memory gets allocated but returning nil there frees it up right away, right?